components/files/file_upload/file_upload.py

```diff
@@ -45,25 +52,66 @@ def to_json(self):
             "upload-file-size" : self.file_size,
         }
 
-def upload_file(a_file_upload_bean, a_file_rdm_bean, a_server_config, a_site_id, a_domain, a_headers):
+# Yield parts of data from file pointer until EOF. This avoids reading large files completely into memory.
+# This function is used to chunk the data if its size exceeds the maximum part size accpeted by the 
+# Sitelink3D v2 file service (and other services that accept file such as designfile)
+def read_parts(a_file_ptr, a_part_size=10485760):
+    while True:
+        part = a_file_ptr.read(a_part_size)
+        if not part:
+            break
+        yield part 
+
+def upload_file_multipart(a_server_config, a_site_id, a_file_upload_bean, media_type, encoding_type, a_headers):

```

Comment:
media_type and encoding_type are not used anymore

Reply:
removed.